org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/themes/AbstractThemeManager.java

```diff
@@ -1,157 +1,134 @@
- *  are made available under the terms of the Eclipse Public License v1.0
- *  which accompanies this distribution, and is available at
- *  http://www.eclipse.org/legal/epl-v10.html
- *
- *  Contributors:
- *  Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
- */
-package org.eclipse.tm4e.ui.internal.themes;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-import org.eclipse.core.runtime.preferences.IEclipsePreferences;
-import org.eclipse.core.runtime.preferences.InstanceScope;
-import org.eclipse.tm4e.ui.themes.ITheme;
-import org.eclipse.tm4e.ui.themes.IThemeAssociation;
-import org.eclipse.tm4e.ui.themes.IThemeManager;
-import org.eclipse.tm4e.ui.themes.ThemeAssociation;
-
-/**
- * TextMate theme manager implementation.
- *
- */
-public abstract class AbstractThemeManager implements IThemeManager {
-
-	// Theme for E4 CSS Engine
-	private static final String E4_CSS_THEME_PREFERENCE_ID = "org.eclipse.e4.ui.css.swt.theme"; //$NON-NLS-1$
-	public static final String E4_THEME_ID = "themeid"; //$NON-NLS-1$
-	private final static String E4_DARK = "org.eclipse.e4.ui.css.theme.e4_dark";
-
-	private final Map<String /* theme id */ , ITheme> themes;
-	private final ThemeAssociationRegistry themeAssociationRegistry;
-
-	public AbstractThemeManager() {
-		this.themes = new LinkedHashMap<>();
-		this.themeAssociationRegistry = new ThemeAssociationRegistry();
-	}
-
-	@Override
-	public void registerTheme(ITheme theme) {
-		themes.put(theme.getId(), theme);
-	}
-
-	@Override
-	public void unregisterTheme(ITheme theme) {
-		themes.remove(theme.getId());
-	}
-
-	@Override
-	public ITheme getThemeById(String themeId) {
-		return themes.get(themeId);
-	}
-
-	@Override
-	public ITheme[] getThemes() {
-		Collection<ITheme> themes = this.themes.values();
-		return themes.toArray(new ITheme[themes.size()]);
-	}
-
-	@Override
-	public ITheme getDefaultTheme() {
-		boolean dark = isDarkEclipseTheme();
-		return getDefaultTheme(dark);
-	}
-
-	public ITheme getDefaultTheme(boolean dark) {
-		for (ITheme theme : this.themes.values()) {
-			if (theme.isDark() == dark && theme.isDefault()) {
-				return theme;
-			}
-		}
-		return null;
-	}
-
-	@Override
-	public ITheme[] getThemes(boolean dark) {
-		Collection<ITheme> themes = this.themes.values();
-		return themes.stream().filter(theme -> {
-			return theme.isDark() == dark;
-		}).collect(Collectors.toList()).toArray(new ITheme[0]);
-	}
-
-	@Override
-	public String getPreferenceE4CSSThemeId() {
-		IEclipsePreferences preferences = getPreferenceE4CSSTheme();
-		return preferences != null ? preferences.get(E4_THEME_ID, null) : null;
-	}
-
-	@Override
-	public boolean isDarkEclipseTheme() {
-		return isDarkEclipseTheme(getPreferenceE4CSSThemeId());
-	}
-
-	@Override
-	public boolean isDarkEclipseTheme(String eclipseThemeId) {
-		return E4_DARK.equals(eclipseThemeId);

```

Comment:
Instead of all this change, to fix the initial request, we can just change it to `eclipseThemeId != null && eclipseThemeId.toLowerCase().contains("dark")`.

Reply:
I think we can close that PR because it's already in #176